appium-tests/android/android.spec.js

```diff
@@ -0,0 +1,338 @@
+        }
+        return screenshotHelper
+            .saveScreenshot(driver)
+            .finally(function() {
+                throw 'rethrow';
+            });
+    }
+
+    function enterTest() {
+        return driver
+            // trying to determine where we are
+            .context(webviewContext)
+            .fail(function (error) {
+                return fail(error);
+            })
+            .elementById('info')
+            .then(function () {
+                return driver; //we're already on the test screen
+            }, function () {
+                return driver
+                    .elementById('middle')
+                    .then(function () {
+                        return driver
+                            // we're on autotests page, we should go to start page
+                            .execute('window.location = "../index.html"')
+                            .sleep(5000)
+                            .fail(function () {
+                                errorFlag = true;
+                                throw 'Couldn\'t find start page.';
+                            });
+                    }, function () {
+                        return; // no-op
+                    })
+                    // unknown starting page: no 'info' div
+                    // adding it manually
+                    .execute('var info = document.createElement("div"); ' +
+                             'info.id = "info"; ' +
+                             'document.body.appendChild(info);');
+            })
+            .sleep(5000);
+    }
+
+    function getDriver() {
+        driver = wdHelper.getDriver('Android');
+        return driver;
+    }
+
+    function checkStopFlag() {
+        if (stopFlag) {
+            fail('Something went wrong: the stopFlag is on. Please see the log for more details.');
+        }
+        return stopFlag;
+    }
+
+    beforeEach(function () {
+        jasmine.addMatchers({
+            toFailWithMessage: function () {
+                return {
+                    compare: function (actual, msg) {
+                        console.log('Failing with message: ' + msg);
+                        var result = {
+                            pass: false,
+                            message: msg
+                        };
+                        // status 6 means that we've lost the session
+                        // status 7 means that Appium couldn't find an element
+                        // both these statuses mean that the test has failed but
+                        // we should try to recreate the session for the following tests
+                        if (msg.indexOf('Error response status: 6') >= 0 ||
+                            msg.indexOf('Error response status: 7') >= 0) {
+                            errorFlag = true;
+                        }
+                        return result;
+                    }
+                };
+            },
+            toContain: function () {
```

Comment:
There is a `toContain` matcher in [Jasmine's standard matchers][matchers].

[matchers]: http://jasmine.github.io/edge/introduction.html#section-Included_Matchers